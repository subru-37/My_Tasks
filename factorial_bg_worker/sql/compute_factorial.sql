SELECT compute_factorial(0);
SELECT compute_factorial(1);
SELECT compute_factorial(5);
SELECT compute_factorial(10);
SELECT compute_factorial(20);
SELECT compute_factorial(21);  
SELECT compute_factorial(-1);  
SELECT compute_factorial(100); 
SELECT compute_factorial(50); 
