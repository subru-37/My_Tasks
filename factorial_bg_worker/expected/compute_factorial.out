SELECT compute_factorial(0);
 compute_factorial 
-------------------
                 1
(1 row)

SELECT compute_factorial(1);
 compute_factorial 
-------------------
                 1
(1 row)

SELECT compute_factorial(5);
 compute_factorial 
-------------------
               120
(1 row)

SELECT compute_factorial(10);
 compute_factorial 
-------------------
           3628800
(1 row)

SELECT compute_factorial(20);
  compute_factorial  
---------------------
 2432902008176640000
(1 row)

SELECT compute_factorial(21);  
ERROR:  Integer overflow
SELECT compute_factorial(-1);  
ERROR:  Negative numbers are not allowed
SELECT compute_factorial(100); 
ERROR:  Integer overflow
SELECT compute_factorial(50); 
ERROR:  Integer overflow
